ProcessLib/SmallDeformationWithLIE/LocalAssembler/SmallDeformationLocalAssemblerMatrix-impl.h

```diff
@@ -0,0 +1,196 @@
+
+        _secondary_data.N[ip] = sm.N;
+    }
+}
+
+
+template <typename ShapeFunction, typename IntegrationMethod,
+          int DisplacementDim>
+void SmallDeformationLocalAssemblerMatrix<ShapeFunction, IntegrationMethod,
+                                    DisplacementDim>::
+assembleWithJacobian(
+    double const t,
+    std::vector<double> const& local_x,
+    std::vector<double> const& /*local_xdot*/,
+    const double /*dxdot_dx*/, const double /*dx_dx*/,
+    std::vector<double>& /*local_M_data*/,
+    std::vector<double>& /*local_K_data*/,
+    std::vector<double>& local_b_data,
+    std::vector<double>& local_Jac_data)
+{
+    assert (_element.getDimension() == DisplacementDim);
+
+    auto const local_matrix_size = local_x.size();
+
+    auto local_Jac = MathLib::createZeroedMatrix<StiffnessMatrixType>(
+        local_Jac_data, local_matrix_size, local_matrix_size);
+
+    auto local_b = MathLib::createZeroedVector<NodalDisplacementVectorType>(
+        local_b_data, local_matrix_size);
+
+    unsigned const n_integration_points =
+        _integration_method.getNumberOfPoints();
+
+    SpatialPosition x_position;
+    x_position.setElementID(_element.getID());
+
+    for (unsigned ip = 0; ip < n_integration_points; ip++)
+    {
+        x_position.setIntegrationPoint(ip);
+        auto const& wp = _integration_method.getWeightedPoint(ip);
+        auto const& detJ = _ip_data[ip]._detJ;
+        auto const& integralMeasure = _ip_data[ip]._integralMeasure;
+
+        auto const& B = _ip_data[ip]._b_matrices;
+        auto const& eps_prev = _ip_data[ip]._eps_prev;
+        auto const& sigma_prev = _ip_data[ip]._sigma_prev;
+
+        auto& eps = _ip_data[ip]._eps;
+        auto& sigma = _ip_data[ip]._sigma;
+        auto& C = _ip_data[ip]._C;
+        auto& material_state_variables =
+            *_ip_data[ip]._material_state_variables;
+
+        eps.noalias() =
+            B *
+            Eigen::Map<typename BMatricesType::NodalForceVectorType const>(
+                local_x.data(), ShapeFunction::NPOINTS * DisplacementDim);
+
+        if (!_ip_data[ip]._solid_material.computeConstitutiveRelation(
+                t, x_position, _process_data.dt, eps_prev, eps, sigma_prev,
+                sigma, C, material_state_variables))
+            OGS_FATAL("Computation of local constitutive relation failed.");
+
+        local_b.noalias() -=
+            B.transpose() * sigma * detJ * wp.getWeight() * integralMeasure;
+        local_Jac.noalias() +=
+            B.transpose() * C * B * detJ * wp.getWeight() * integralMeasure;
+    }
+}
+
+
+template <typename ShapeFunction, typename IntegrationMethod,
+          int DisplacementDim>
+void SmallDeformationLocalAssemblerMatrix<ShapeFunction, IntegrationMethod,
+                                    DisplacementDim>::
+postTimestepConcrete(std::vector<double> const& /*local_x*/)
+{
+    const int n = 3;
+    std::valarray<double> ele_stress(0.0, n);
```

Comment:
Minor: I'd prefer Eigen types here instead of `valarray`, because the former are used everywhere anyways.